libs/ngrx-toolkit/src/lib/with-resource.spec.ts

```diff
@@ -0,0 +1,662 @@
+
+          it('has error status when error', async () => {
+            const { store, addressResolver } = setupWithNamedResource();
+
+            addressResolver.resolve.mockRejectedValue(new Error('Error'));
+            patchState(store, { id: 1 });
+            await wait();
+
+            expect(store.addressStatus()).toBe('error');
+            expect(() => store.addressValue()).toThrow();
+            expect(store.addressError()).toBeInstanceOf(Error);
+            expect(store.addressIsLoading()).toBe(false);
+            expect(store.addressHasValue()).toBe(false);
+          });
+
+          it('has local once updated', async () => {
+            const { store, addressResolver } = setupWithNamedResource();
+
+            addressResolver.resolve.mockResolvedValue(venice);
+            patchState(store, { id: 1 });
+
+            await wait();
+            patchState(store, ({ addressValue }) => ({
+              addressValue: addressValue
+                ? { ...addressValue, country: 'Italia' }
+                : undefined,
+            }));
+
+            expect(store.addressStatus()).toBe('local');
+            expect(store.addressValue()?.country).toBe('Italia');
+            expect(store.addressError()).toBeUndefined();
+            expect(store.addressIsLoading()).toBe(false);
+            expect(store.addressHasValue()).toBe(true);
+          });
+
+          it('can also reload by resource name', async () => {
+            const { store, addressResolver } = setupWithNamedResource();
+
+            addressResolver.resolve.mockResolvedValueOnce(venice);
+            patchState(store, { id: 1 });
+            await wait();
+            expect(store.addressStatus()).toBe('resolved');
+            store.reload();
+            expect(store.addressStatus()).toBe('reloading');
+          });
+        });
+      });
+
+      describe('override protection', () => {
+        const warningSpy = jest.spyOn(console, 'warn');
+
+        afterEach(() => {
+          warningSpy.mockClear();
+        });
+
+        for (const memberName of [
+          //TODO wait for https://github.com/ngrx/platform/pull/4932
+          // 'value',
+          'status',
+          'error',
+          'isLoading',
+          '_reload',
+          'hasValue',
+        ]) {
+          it(`warns if ${memberName} is not a member of the store`, () => {
+            const Store = signalStore(
+              { providedIn: 'root' },
+              withProps(() => ({ [memberName]: true })),
+              withResource(() =>
+                resource({ loader: () => Promise.resolve(1) }),
+              ),
+            );
+
+            TestBed.inject(Store);
+
+            expect(warningSpy).toHaveBeenCalledWith(
+              '@ngrx/signals: SignalStore members cannot be overridden.',
+              'Trying to override:',
+              memberName,
+            );
+          });
+        }
+
+        //TODO wait for https://github.com/ngrx/platform/pull/4932
```

Comment:
Same question as above - is this blocking for the PR or just something to keep in mind?

Reply:
Closing that one since we are not waiting